pkg/cfgstruct/bind.go

```diff
@@ -147,10 +164,26 @@ func bindConfig(flags FlagSet, prefix string, val reflect.Value,
 			default:
 				panic(fmt.Sprintf("invalid field type: %s", field.Type.String()))
 			}
+			if setup {
+				setSetupAnnotation(flags, flagname)
+			}
 		}
 	}
 }
 
+func setSetupAnnotation(flagset interface{}, name string) {
+	flags, ok := flagset.(*pflag.FlagSet)
+	if !ok {
+		return
+	}
+
+	err := flags.SetAnnotation(name, "setup", []string{"true"})
+	if err != nil {
+		fmt.Println("unable to set annotation:", err)
```

Comment:
should we use a logger?

Reply:
I think we aren't using logger in commands, at least for now. tbh I'm not sure if this should be logged at all.

Reply:
done